pandas/_typing.py

```diff
@@ -83,8 +83,11 @@
     # Name "npt._ArrayLikeInt_co" is not defined  [name-defined]
     NumpySorter = Optional[npt._ArrayLikeInt_co]  # type: ignore[name-defined]
 
+    from typing_extensions import TypeGuard  # pyright: reportUnusedImport = false
+
 else:
     npt: Any = None
+    TypeGuard: Any = None
```

Comment:
`TypeGuard = bool`? In my understanding a `TypeGuard` is like a bool that ensure that another has some type…

Reply:
This should probably not be outside of the `if TYPE_CHECKING` section?